back/engines/commercial/smart_groups/app/lib/smart_groups/rules/email.rb

```diff
@@ -50,7 +50,15 @@ def initialize(predicate, value)
       self.value = value
     end
 
-    def filter(users_scope)
+    def cachable?
+      true
+    end
+
+    def cache_key_fragments
+      [User.all.cache_key_with_version, predicate, value]
+    end
+
+    def query(users_scope)
```

Comment:
Should `query` become a private method in all rules?

Reply:
Yes and done.